src/types/datetime.rs

```diff
@@ -487,6 +488,32 @@ pub fn timezone_utc(py: Python<'_>) -> &PyTzInfo {
     unsafe { &*(ensure_datetime_api(py).TimeZone_UTC as *const PyTzInfo) }
 }
 
+/// Equivalent to `datetime.timezone` constructor
+///
+/// Use [`timezone_from_offset_and_name`] for a named timezones.
+pub fn timezone_from_offset<'a>(py: Python<'a>, offset: &PyDelta) -> PyResult<&'a PyTzInfo> {
```

Comment:
My main question here is why this has to become new public API if it is mainly meant as extension point to differentiate between accessing unlimited and limited API? Wouldn't `pub(crate)` suffice for that purpose?

Reply:
Yes, indeed. Done.